src/runtime/refiner.ts

```diff
@@ -1671,6 +2506,7 @@ export abstract class RefinementExpressionVisitor<T> {
   abstract visitQueryArgumentPrimitive(expr: QueryArgumentPrimitive): T;
   abstract visitBuiltIn(expr: BuiltIn): T;
   abstract visitNumberPrimitive(expr: NumberPrimitive): T;
+  abstract visitBigIntPrimitive(expr: BigIntPrimitive): T;
   abstract visitBooleanPrimitive(expr: BooleanPrimitive): T;
   abstract visitTextPrimitive(expr: TextPrimitive): T;
 }
```

Comment:
Not for this PR - this file is getting pretty big; might be worth splitting it if it's logically sensible to do so.

Reply:
100% Probably should have split it a while ago, will have to follow this up.